### DISCUSSION TITLE
Is it safe to assume that conv layers without a `batch` argument for `forward` are safe to use with batched graphs?

### DISCUSSION THREAD

**Discussion by callumm-graphcore** [pyg-team/pytorch_geometric#5134] [Q&A] [answered]
Hello,

I'm researching PyG because we are planning on writing tutorials on it for my company's platform. I've noticed that some convolutional layers take an optional `batch` argument to their `forward` method containing a batch vector. Does this mean that we can assume that the layers that don't allow you to provide such an argument can be safely used with "batched" graphs?

With thanks,
Callum

**Comment by rusty1s** [comment, marked as answer]
Yes. Any GNN layer without a `batch` argument can be used for single graphs and batched graphs.

**Reply by callumm-graphcore** [reply]
Great, thanks Matthias!
